samza-core/src/main/java/org/apache/samza/zk/ZkUtils.java

```diff
@@ -65,12 +66,10 @@
   private volatile String ephemeralPath = null;
   private final ZkKeyBuilder keyBuilder;
   private final int connectionTimeoutMs;
-  private ZkJobCoordinatorMetrics metrics;
+  private final ZkJobCoordinatorMetrics metrics;
 
   public ZkUtils(ZkKeyBuilder zkKeyBuilder, ZkClient zkClient, int connectionTimeoutMs) {
-    this.keyBuilder = zkKeyBuilder;
-    this.connectionTimeoutMs = connectionTimeoutMs;
-    this.zkClient = zkClient;
+    this (zkKeyBuilder, zkClient, connectionTimeoutMs, new ZkJobCoordinatorMetrics(new MetricsRegistryMap()));
```

Comment:
Q1: 
No, it's used at other places as well(`ZkCoordinationServiceFactory`).

Q2:

Instantiation of the MetricsRegistry don’t change with any configuration parameter provided by client of this constructor(In essence not tunable by any configuration). 

This is more like convenient shorthand, if there’re multiple places if this is called and all of them have to do the same instantiation again and again, I prefer moving it one level up here to minimize redundancy . 

What is the disadvantage of having like this and why should the client be aware of this dependency (it doesn’t affect normal operability of any api)?? 

If by mistake, this param is set to null, some ZkUtils api's will break with NPE. To prevent that from happening, we can  delete the other constructor.

Please let me know your thoughts or if i’m missing something.